### PR TITLE
PHPStan: Add support for multiline messages

### PR DESCRIPTION
**Pull request by greg0ire**
When using PHPStan, you might get multiline messages in some cases:

```
/home/greg/dev/doctrine-migrations/tests/Doctrine/Migrations/Tests/Metadata/Storage/ExistingTableMetadataStorageTest.php:83:Fetching class constant class of deprecated class Doctrine\DBAL\Connections\MasterSlaveConnection:
Use PrimaryReadReplicaConnection instead
/home/greg/dev/doctrine-migrations/tests/Doctrine/Migrations/Tests/Metadata/Storage/TableMetadataStorageTest.php:200:Instantiation of deprecated class Doctrine\DBAL\Driver\PDOSqlite\Driver:
Use {@link PDO\SQLite\Driver} instead.
/home/greg/dev/doctrine-migrations/tests/Doctrine/Migrations/Tests/Tools/Console/Command/StatusCommandTest.php:77:Call to method compare() of deprecated class Doctrine\DBAL\Version:
Refrain from checking the DBAL version at runtime.
```

Without this patch, the messages get truncated and you only get the first line, so just `Instantiation of deprecated class Doctrine\DBAL\Driver\PDOSqlite\Driver:`

With it, you can get the whole message. (`Instantiation of deprecated class Doctrine\DBAL\Driver\PDOSqlite\Driver:
Use {@link PDO\SQLite\Driver} instead.`)

This is achieved by switching to JSON, which makes it much easier to
avoid confusion between an error message and the next one. It also
spares us from having to deal with regular expressions, and eliminates
some edge cases that no longer need to be tested.

### CODE REVIEW

File: test/handler/test_phpstan_handler.vader [hunk 1]
```diff
@@ -38,37 +40,5 @@ Execute(Output with some errors should be parsed correctly):
...
-  \ [
-  \   {
-  \     'lnum': 9,
-  \     'text': 'Access to an undefined property Test::$var.',
-  \     'type': 'E'
-  \   }
-  \ ],
-  \ ale_linters#php#phpstan#Handle(bufnr(''), [
-  \   '/phpstan-test-files/foo/test.inc:9:Access to an undefined property Test::$var.',
+  \ '{"totals":{"errors":0,"file_errors":11},"files":{"'.expand('%:p').'":{"errors":3,"messages":[{"message":"Call to method format() on an unknown class DateTimeImutable.","line":9,"ignorable":true},{"message":"Sample message.","line":16,"ignorable":true},{"message":"Invalid command testCommand.","line":192,"ignorable":true}]}}}',
```
Comment (greg0ire) [inline test/handler/test_phpstan_handler.vader:43]: Apparently this does not work as expected on Windows, not sure how to debug it.

Comment (greg0ire) [reply test/handler/test_phpstan_handler.vader:43]: When I try troubleshooting this with `echom`, it hangs forever :disappointed:

Comment (greg0ire) [reply test/handler/test_phpstan_handler.vader:48]: This is what the dictionary looks like on Windows:

`{"C:\testplugin\testphpstan-test-files\foo\test.php":{"messages":[{"line":9,"message":"Call to method format() on an unknown class DateTimeImutable.","ignorable":true},{"line":16,"message":"Sample message.","ignorable":true},{"line":192,"message":"Invalid command testCommand.","ignorable":true}],"errors":3}}`

Notice the missing backslash between `C:\testplugin\test` and `phpstan-test-files\foo\test.php`

**Comment by greg0ire** [comment]
ping @w0rp I've been using this for almost 2 months now, no issues :slightly_smiling_face:

**Review by hsanson — APPROVED** [review]
Looks good to me and json output seems to have been supported for long time.
